app/models/concerns/profile_searching.rb

```diff
@@ -35,8 +35,22 @@ module ProfileSearching
       where(external: external)
     }
     scope :has_test_results, lambda { |has_test_results = true|
-      test_results = TestResult.select(:profile_id).distinct
-      has_test_results && where(id: test_results) || where.not(id: test_results)
+      # the use of default scope is to bypass Pundit
+      # and avoid the situation with bind parameters
+      # that must have a different order in the resulting query
+      # of a scoped_search
+      policy_test_results = default_scoped.joins(policy_object: :test_results)
+
+      internal = arel_table[:external].eq(false).and(
+        arel_table[:id].in(policy_test_results.select(:id).arel)
+      )
+
+      external = arel_table[:external].eq(true).and(
+        arel_table[:id].in(TestResult.default_scoped.select(:profile_id).arel)
+      )
+
+      cond = internal.or(external)
```

Comment:
I think we can use the old `has_test_results` here instead of `external`.

Reply:
I've completely dropped the `external` condition.